### PR TITLE
add testcontainer as integration testing tool

### LINKED ISSUE #103: Research testcontainer as integration testing tool.
**What would you like to be added**:

**Why is this needed**:

https://github.com/testcontainers/testcontainers-go

https://www.testcontainers.org/

### PR DESCRIPTION
**Pull request by dongzl**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #103 .

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by AlexStocks — APPROVED** [review]

**Review by baerwang — APPROVED** [review]

File: test/testcontainer_mysql_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		log.Error("Failed to setup MySQL container")
+		panic(err)
+	}
+	os.Exit(m.Run())
+}
+
+func TestSelect_Integration(t *testing.T) {
+	rows, err := db.Query(`SELECT uid, name, score, nickname FROM student_0001 where uid = ?`, 1)
+	assert.NoErrorf(t, err, "select row error: %v", err)
+
```
Comment (jjeffcaii) [inline test/testcontainer_mysql_test.go:60]: add `defer rows.Close()`

File: test/testcontainer_mysql_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		name     string
+		score    float64
+		nickname string
+	)
+
+	if rows.Next() {
+		err = rows.Scan(&uid, &name, &score, &nickname)
+		if err != nil {
+			t.Error(err)
+		}
```
Comment (jjeffcaii) [inline test/testcontainer_mysql_test.go:72]: Use `assert.NoError(...)` instead.

**Comment by jjeffcaii** [comment]
I think the test source file (test_container.go) should be renamed to `testcontainer_common_test.go`, for example. Because it's for test scope only.

**Review by jjeffcaii — APPROVED** [review]
LGTM

**Review by Penglq — APPROVED** [review]

**Pull request merged**